Expensify/react-native-live-markdown
# Inline images feature

### Details

This PR adds the ✨ `inline image preview feature` ✨ to the Live Markdown Input on the web.


https://github.com/user-attachments/assets/a7a0be95-a773-4d8f-83d8-7a6dc3d113c8



Change log:
- added inline image previews when inline image tags are present in the input
- added support for style and resolution customization
- added loading indicator spinners that appear when the image previews are being loaded
- added `block` element wrappers, that wraps the inline image markdown and text that is before it. Thanks to this structure change, the image preview appears as a block below the markdown 

### Related Issues

[GH_LINK](https://github.com/Expensify/App/issues/40181#issue-2240640829)

### Manual Tests

1. Insert inline image markdown text into the input. I.e
```
![demo image](https://picsum.photos/200/300)
```
2. Verify cursor positioning and text alignment while writing text around the image preview
3. Verify image loading spinner animation and behavior
4. Verify inline image styles, and try changing the styles passed to the component. Especially the `max-height` and `maxWidth` of the preview
5. Verify if text after the preview can be removed on `Backspace` press
6. Verify if the Live Markdown Input behaves normally in other situations that come to your mind

### Linked PRs
